react-components/src/metadata/admin/AttributeEnumEditGrid.js

```diff
@@ -0,0 +1,269 @@
+import TableHead from "@material-ui/core/TableHead";
+import TableRow from "@material-ui/core/TableRow";
+import { withStyles } from "@material-ui/core/styles";
+
+import ContentRemove from "@material-ui/icons/Delete";
+import ContentEdit from "@material-ui/icons/Edit";
+
+const normalizeDefault = (fields, change, is_default) => {
+    if (is_default) {
+        fields.forEach((field, index) => {
+            let attrEnum = fields.get(index);
+            if (attrEnum.is_default) {
+                change(field, {...attrEnum, is_default: false});
+            }
+        });
+    }
+
+    return is_default;
+};
+
+class AttributeEnumEditDialog extends Component {
+    normalizeDefaultField = (is_default) => {
+        const { fields, change } = this.props;
+        return normalizeDefault(fields, change, is_default);
+    };
+
+    render() {
+        const { open, intl, field, error, onClose } = this.props;
+
+        const dialogID = build(ids.METADATA_TEMPLATE_FORM, field, ids.DIALOG);
+        const dialogTitleID = build(dialogID, ids.TITLE);
+
+        return (
+            <Dialog
+                open={open}
+                disableBackdropClick
+                disableEscapeKeyDown
+                aria-labelledby={dialogTitleID}
+            >
+                <DialogTitle id={dialogTitleID}>{getMessage("dialogTitleEditEnumValue")}</DialogTitle>
+                <DialogContent>
+                    <Field name={`${field}.value`}
+                           label={formatMessage(intl, "value")}
+                           id={build(dialogID, ids.ENUM_VALUE)}
+                           required={true}
+                           autoFocus
+                           margin="dense"
+                           component={FormTextField}
+                    />
+                    <Field name={`${field}.is_default`}
+                           label={formatMessage(intl, "enumDefaultLabel")}
+                           id={build(dialogID, ids.ENUM_VALUE_DEFAULT)}
+                           color="primary"
+                           component={FormCheckbox}
+                           normalize={this.normalizeDefaultField}
+                    />
+                </DialogContent>
+                <DialogActions>
+                    <Button id={build(dialogID, ids.BUTTONS.CLOSE)}
+                            color="primary"
+                            disabled={!!error}
+                            onClick={onClose}
+                    >
+                        {getMessage("done")}
+                    </Button>
+                </DialogActions>
+            </Dialog>
+        );
+    }
+}
+
+class AttributeEnumEditGrid extends Component {
+    constructor(props) {
+        super(props);
+
+        this.newEnumCount = 1;
+
+        this.state = {
+            selected: -1,
+            editingEnumIndex: -1,
+        };
+    }
+
+    newEnumValue = () => `New value ${this.newEnumCount++}`;
```

Comment:
Yep, this is the default new value for enum options added to the form by an admin, so it will be displayed in the form until the admin changes it.

I'll add it to the i18n messages 👍